src/rvsdg/tree_unique/to_tree.rs

```diff
@@ -55,37 +61,39 @@ impl<'a> RegionTranslator<'a> {
         res
     }
 
-    /// Build a translator and translate
-    /// the operands to the tree encoding.
-    /// Produces a tree-encoded term that evaluates
-    /// to a tuple containing results.
-    fn translate(num_args: usize, nodes: &'a Vec<RvsdgBody>, results: Vec<Operand>) -> Expr {
-        let mut translator = RegionTranslator {
+    /// Make a new translator for a region with
+    /// num_args and the given nodes.
+    fn new(num_args: usize, nodes: &'a Vec<RvsdgBody>) -> RegionTranslator {
+        RegionTranslator {
             num_args,
             bindings: Vec::new(),
             index_of: HashMap::new(),
             nodes,
-        };
-
-        let mut result_indices = Vec::new();
-        for result in results {
-            result_indices.push(translator.translate_operand(result));
         }
+    }
 
-        let mut expr = sequence(result_indices.iter().map(|i| get(arg(), *i)).collect());
+    /// Wrap the given expression in all the
+    /// bindings that have been generated.
+    fn build_translation(&self, inner: Expr) -> Expr {
+        let mut expr = inner;
 
-        for binding in translator.bindings.into_iter().rev() {
-            expr = cbind(binding, expr);
+        for binding in self.bindings.iter().rev() {
+            expr = cbind(binding.clone(), expr);
         }
         expr
     }
 
-    fn translate_operand(&mut self, operand: Operand) -> usize {
+    /// Returns a pure expression (e.g. `getarg(0)`) that
+    /// returns the value for this operand.
+    /// The value of the operand is let-bound
+    /// and the expression refers to it.
+    fn translate_operand(&mut self, operand: Operand) -> Expr {
         match operand {
-            Operand::Arg(index) => index,
-            Operand::Id(id) => self.translate_node(id),
-            Operand::Project(_id, _indexx) => {
-                todo!("Doesn't handle subregions yet");
+            Operand::Arg(index) => getarg(index),
+            Operand::Id(id) => getarg(self.translate_node(id)),
+            Operand::Project(p_index, id) => {
+                let index = self.translate_node(id);
```

Comment:
Why is this called index? Isn't this a region?

Reply:
The whole region becomes a let-bound tuple, so this index is the index of that tuple in the environment 
Added some comments that explain this better